### PR TITLE
Add 2022 year in review blog post

### PR DESCRIPTION
**Pull request by darrenburns**
image

### CODE REVIEW

**Comment by darrenburns** [comment]
TODO @darrenburns : Link to rich-pixels - mentioned library but not linked to it or said name of it.

File: docs/blog/posts/darren-year-in-review.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+---
+draft: true
```
Comment (darrenburns) [inline docs/blog/posts/darren-year-in-review.md:2]: Have left this as a draft for now.

File: docs/blog/posts/darren-year-in-review.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+While working on [Shira](https://github.com/darrenburns/shira) (a proof-of-concept, terminal-based Python object explorer), I wrote some autocompleting dropdown functionality.
+
+![shira-demo.gif](../images/darren-year-in-review/shira-demo.gif)
+
+Textual forgoes the z-index concept from browser CSS and instead uses a “named layer” system. Using the `layers` property you can defined an ordered list of named layers, and using the `layer` property, you can assign a descendant widget to one of those layers.
+
+By creating a new layer above all others and assigning a widget to that layer, we can ensure that widget is painted above everything else.
+
+In order to determine where to place the dropdown, we can track the current value in the dropdown by `watch`ing the reactive input “value” inside the Input widget. This method will be called every time the `value` of the Input changes, and we can use this hook to amend the position of our dropdown position to accommodate for the length of the input value.
```
Comment (darrenburns) [inline docs/blog/posts/darren-year-in-review.md:68]: I feel this part might be a bit jarring - I randomly start going into a bit too much detail. Maybe I should talk a bit more about Shira and the functionality of the autocomplete library at a high level instead.

Comment (willmcgugan) [reply docs/blog/posts/darren-year-in-review.md:68]: Think I'll be publishing on Sunday. I think this is great, but feel free to PR edits before then if you like.

**Review by willmcgugan — APPROVED** [review]

File: docs/blog/posts/darren-year-in-review.md (hunk 2 above)
Comment (darrenburns) [reply docs/blog/posts/darren-year-in-review.md:68]: @willmcgugan Should I set `draft: false` now or will you do it?

Comment (willmcgugan) [reply docs/blog/posts/darren-year-in-review.md:68]: I'll do it.

**Pull request merged**